components/tidb_query_aggr/src/impl_max_min.rs

```diff
@@ -186,6 +186,94 @@ where
     }
 }
 
+#[derive(Debug, AggrFunction)]
+#[aggr_function(state = AggFnStateExtremumForEnum::<E>::new())]
+pub struct AggFnExtremumForEnum<E>
+where
+    E: Extremum,
+    VectorValue: VectorValueExt<Enum>,
+{
+    _phantom: std::marker::PhantomData<E>,
+}
+
+impl<E> AggFnExtremumForEnum<E>
+where
+    E: Extremum,
+    VectorValue: VectorValueExt<Enum>,
+{
+    fn new() -> Self {
+        Self {
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct AggFnStateExtremumForEnum<E>
+where
+    E: Extremum,
+    VectorValue: VectorValueExt<Enum>,
+{
+    extremum: Option<Enum>,
+    _phantom: std::marker::PhantomData<E>,
+}
+
+impl<E> AggFnStateExtremumForEnum<E>
+where
+    E: Extremum,
+    VectorValue: VectorValueExt<Enum>,
+{
+    pub fn new() -> Self {
+        Self {
+            extremum: None,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// # Notes
+    ///
+    /// For MAX(), MySQL currently compares ENUM and SET columns by their string value rather
+    /// than by the string's relative position in the set. This differs from how ORDER BY
+    /// compares them.
+    ///
+    /// ref: https://dev.mysql.com/doc/refman/5.7/en/aggregate-functions.html#function_max
+    #[inline]
+    fn update_concrete(&mut self, _ctx: &mut EvalContext, value: Option<EnumRef>) -> Result<()> {
+        let extreme_ref = self
+            .extremum
+            .as_ref()
+            .map(|x| EnumRef::from_owned_value(unsafe { std::mem::transmute(x) }));
+
+        if value.is_some()
+            && (self.extremum.is_none()
+                || extreme_ref
+                    .unwrap()
+                    .to_string()
```

Comment:
Is it necessary to allocate string for every comparison?

Reply:
This is current limitation of copr framework. We must store owned value inside aggregator. But here I think we could store `Enum` instead of `String`, to improve efficiency.

Reply:
Maybe we can compare the underlying bytes directly.

Should we need to care about string collections here?

Reply:
I agree. This is not necessary. Can we add some interface like `to_str()` on `EnumRef`?

Reply:
Let's give it a try.

Reply:
By the way, I think it's necessary to take collation into consideration here. And I suggest to implement it in separate PR.

Reply:
as_str() maybe better?

Reply:
Agree. We may use `as_str()` to get `&str`.

Reply:
> By the way, I think it's necessary to take collation into consideration here. And I suggest to implement it in separate PR.

OK, I wiil add a TODO here.

Reply:
One more thing, build str from `u8` without convertion requires utf-8 check, should we just panic here or use ` String::from_utf8_lossy` to convert invliad bytes to `U+FFFD`?

Reply:
I think we could return `Err` here.